plugins/TrollManagement/tests/TrollManagementTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @author David Barbier<david.barbier@vanillaforums.com>
+ * @copyright 2009-2021 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\TrollManagement;
+
+use Garden\Container\Container;
+use UserModel;
+use RoleModel;
+use BanModel;
+use VanillaTests\APIv0\TestDispatcher;
+use VanillaTests\SetupTraitsTrait;
+use VanillaTests\SiteTestCase;
+use VanillaTests\TestInstallModel;
+
+/**
+ * Class TrollManagementTest
+ */
+class TrollManagementTest extends SiteTestCase {
+    use SetupTraitsTrait;
+
+    /** @var UserModel */
+    protected $userModel;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setUp(): void {
+        parent::setUp();
+        $this->createUserFixtures();
+    }
+
+    /**
+     * Setup routine, run before the test class is instantiated.
+     */
+    public static function setupBeforeClass(): void {
+        self::$addons = ['vanilla', 'trollmanagement'];
+
+        parent::setupBeforeClass();
+    }
+
+    /**
+     * Tests the automatic assignment of the "applicant" status to a new user registration if the maximum amount of
+     * user accounts sharing the same fingerprint is reached.
+     */
+    public function testRegisterApplicant(): void {
+        /** @var \Gdn_Configuration $configuration */
+        $configuration = static::container()->get('Config');
+
+        $configuration->set('TrollManagement.PerFingerPrint.Enabled', true);
+        $configuration->set('TrollManagement.PerFingerPrint.MaxUserAccounts', 3);
+
+        // Ensure all future registered dummy users uses the same fingerprint.
+        $_COOKIE['__vnf'] = 'THISISAFAKEFINGERPRINT';
+
+        // Create 3 dummy accounts. (The third should be automatically set as "applicant")
+        $importedUsers[] = $this->insertDummyUser();
+        $importedUsers[] = $this->insertDummyUser();
+        $importedUsers[] = $this->insertDummyUser();
+
+        // We pull the associated user's roles.
+        foreach ($importedUsers as $importedUser) {
+            $importedUsersRolesIDs[] = $this->userModel->getRoleIDs($importedUser['UserID']);
+        }
+
+        // The FIRST dummy user account is NOT an applicant
+        $this->assertNotContains(RoleModel::APPLICANT_ID, $importedUsersRolesIDs['0']);
+        // The SECOND dummy user account is NOT an applicant
+        $this->assertNotContains(RoleModel::APPLICANT_ID, $importedUsersRolesIDs['1']);
+        // The THIRD dummy user account should an applicant
```

Comment:
This setting is documented in the UI as "maximum allowed number of user accounts tied to a single fingerprint", but if I say the maximum number of accounts a user can have is three, I would expect them to be able to register three accounts. If I'm reading this correctly, we're actually only allowing them to register two accounts.

Reply:
Agreed. The test has been adjusted.